docker-compose/prepareHostEnv.sh

```diff
@@ -147,28 +149,44 @@ copyFiles () {
     echo "Copying needed files to directories"
 
     echo "Artifactory configuration files"
-    if [ "$TYPE" = "pro" ] || [ "$TYPE" = "oss" ]; then
+    if [ "${TYPE}" = "pro" ] || [ "${TYPE}" = "oss" ]; then
         cp -fr ${SCRIPT_DIR}/../files/access ${ROOT_DATA_DIR}/artifactory/
     else
         cp -fr ${SCRIPT_DIR}/../files/access ${ROOT_DATA_DIR}/artifactory/node1/
         cp -fr ${SCRIPT_DIR}/../files/access ${ROOT_DATA_DIR}/artifactory/node2/
     fi
-        # Copy the binarystore.xml which has configuration for no-shared storage
-        if [ "$TYPE" = "ha" ]; then
-            cp -f ${SCRIPT_DIR}/../files/binarystore.xml ${ROOT_DATA_DIR}/artifactory/node1/etc
-        fi
+
+    # Copy the binarystore.xml which has configuration for no-shared storage
+    if [ "${TYPE}" = "ha" ]; then
+        cp -f ${SCRIPT_DIR}/../files/binarystore.xml ${ROOT_DATA_DIR}/artifactory/node1/etc
+    fi
 
     local type=${TYPE}
     if [ ${type} = "ha" ]; then type=ha; fi
 
     echo "Nginx Artifactory configuration"
     cp -fr ${SCRIPT_DIR}/../files/nginx/conf.d/${type}/* ${ROOT_DATA_DIR}/nginx/conf.d/
+}
 
+setPermissions () {
+    # Fix directories permissions only on Linux
+    if [ ${OS_NAME} == "Linux" ]; then
+        echo "Creating ${ARTIFACTORY_USER_NAME} user"
+        if id ${ARTIFACTORY_USER_NAME} 2> /dev/null; then
+            echo "User ${ARTIFACTORY_USER_NAME} already exists"
+        else
+            useradd -M -s /usr/sbin/nologin --uid ${ARTIFACTORY_USER_ID} --user-group ${ARTIFACTORY_USER_NAME} || errorExit "Creating user ${ARTIFACTORY_USER_NAME} failed"
+        fi
+
+        echo "Setting permissions and ownership"
+        chown -R ${ARTIFACTORY_USER_NAME}. ${ROOT_DATA_DIR}/artifactory || errorExit "Setting ownership of ${ROOT_DATA_DIR}/artifactory to ${ARTIFACTORY_USER_ID} failed"
+        chmod -R 777 ${ROOT_DATA_DIR}/nginx || errorExit "Setting permissions 777 on ${ROOT_DATA_DIR}/nginx failed"
```

Comment:
Do we still need this?

Reply:
Since these are examples, and this is the nginx volumes, I thought it's a simple fix...